extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -483,6 +484,14 @@ private static boolean collectDeclaredProviders(List<RestClientBuildItem> restCl
                     }
                 }
             }
+
+            // handle @PartType: we don't know if it's used for writing or reading so we register both
+            for (AnnotationInstance partTypeAnnotation : index.getAnnotations(ResteasyDotNames.RESTEASY_PART_TYPE)) {
+                providersToRegister
+                        .addAll(categorizedReaders.getPossible(MediaType.valueOf(partTypeAnnotation.value().asString())));
```

Comment:
I think an `IllegalArgumentException` may be thrown if the `MediaType` cannot be resolved? Also, it would be nice to call `MediaType.valueOf` just one time

Reply:
Right, fixed!